R/combine_checkboxes.R

```diff
@@ -0,0 +1,192 @@
+#'
+#' @description
+#' [combine_checkboxes()] consolidates multiple checkbox fields in a REDCap data
+#' tibble into a single column. This transformation simplifies analysis by
+#' combining multiple binary columns into a singular and informative labelled
+#' factor column.
+#'
+#' @param supertbl A supertibble generated by [read_redcap()]. Required.
+#' @param tbl The `redcap_form_name` of the data tibble to extract. Required.
+#' @param cols <[`tidy-select`][tidyr_tidy_select]> Checbox columns to combine to
+#' single column. Required.
+#' @param values_to A string specifying the name of the column to combine checkbox
+#' values under. Required.
+#' @param multi_value_label A string specifying the value to be used when multiple
+#' checkbox fields are selected. Default "Multiple".
+#' @param values_fill Value to use when no checkboxes are selected. Default `NA`.
+#' @param raw_or_label Either 'raw' or 'label' to specify whether to use raw coded
+#' values or labels for the options. Default 'label'.
+#' @param keep Logical indicating whether to keep the original checkbox fields in
+#' the output. Default `TRUE`.
+#'
+#' @return A modified supertibble.
+#'
+#' @examples
+#' \dontrun{
+#' supertbl <- read_redcap(redcap_uri, token)
+#' combined_tbl <- combine_checkboxes(
+#'   supertbl = supertbl,
+#'   tbl = "demographics",
+#'   cols = starts_with("race"),
+#'   values_to = "race_combined",
+#'   multi_value_label = "Multiple",
+#'   values_fill = NA
+#' )
+#' }
+#'
+#' @export
+
+combine_checkboxes <- function(supertbl,
+                               tbl,
+                               cols,
+                               values_to,
+                               multi_value_label = "Multiple",
+                               values_fill = NA,
+                               raw_or_label = "label",
+                               keep = TRUE) {
+
+  # Check args ---
+  check_arg_is_supertbl(supertbl, req_cols = c("redcap_data", "redcap_metadata"))
+  check_arg_is_character(tbl, len = 1, any.missing = FALSE)
+  check_arg_is_character(values_to, len = 1, any.missing = FALSE)
+  check_arg_is_character(multi_value_label, len = 1, any.missing = TRUE)
+  check_arg_is_character(values_fill, len = 1, any.missing = TRUE)
+  check_arg_choices(raw_or_label, choices = c("label", "raw"))
+  check_arg_is_logical(keep, len = 1, any.missing = FALSE)
+
+  # Save user cols to enquosure
+  cols_exp <- enquo(cols)
+
+  # Extract tbl from supertbl
+  data_tbl <- supertbl %>%
+    extract_tibble(tbl)
+
+  # Get field names from cols_exp, check that fields exist
+  field_names <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = field_names, expr = cols_exp)
+
+  # Assume the first instrument in the metadata contains IDs
+  # REDCap enforces this constraints, we reflect this in read_redcap -> get_field_to_drop
+  record_id_field <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]$field_name[1]
```

Comment:
Yea, certainly. My original thought about this was when we call this in `read_redcap()` we know we're getting data from the source, but that perhaps it would be safer to go to the metadata for a post-retrieved supertibble. The approach above may not even address it, but it was something I was mulling over.

Will changed to: 

```r
record_id_field <- get_record_id_field(supertbl$redcap_data[[1]])
```

Reply:
With the most recent change `record_id_field` isn't currently needed anymore.